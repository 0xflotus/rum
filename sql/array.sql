set enable_seqscan=off;


CREATE TABLE test_array (
	i int2[]
);

INSERT INTO test_array VALUES ('{}'), ('{0}'), ('{1,2,3,4}'), ('{1,2,3}'), ('{1,2}'), ('{1}');

CREATE INDEX idx_array ON test_array USING rum (i);

EXPLAIN (COSTS OFF) SELECT * FROM test_array WHERE i = '{}';
SELECT * FROM test_array WHERE i = '{}';
SELECT * FROM test_array WHERE i = '{0}';
SELECT * FROM test_array WHERE i = '{1}';
SELECT * FROM test_array WHERE i = '{1,2}';
SELECT * FROM test_array WHERE i = '{2,1}';
SELECT * FROM test_array WHERE i = '{1,2,3,3}';
SELECT * FROM test_array WHERE i = '{0,0}';
SELECT * FROM test_array WHERE i = '{100}';

EXPLAIN (COSTS OFF) SELECT * FROM test_array WHERE i && '{}';
SELECT * FROM test_array WHERE i && '{}';
SELECT * FROM test_array WHERE i && '{1}';
SELECT * FROM test_array WHERE i && '{2}';
SELECT * FROM test_array WHERE i && '{3}';
SELECT * FROM test_array WHERE i && '{4}';
SELECT * FROM test_array WHERE i && '{1,2}';
SELECT * FROM test_array WHERE i && '{1,2,3}';
SELECT * FROM test_array WHERE i && '{1,2,3,4}';
SELECT * FROM test_array WHERE i && '{4,3,2,1}';
SELECT * FROM test_array WHERE i && '{0,0}';
SELECT * FROM test_array WHERE i && '{100}';

EXPLAIN (COSTS OFF) SELECT * FROM test_array WHERE i @> '{}';
SELECT * FROM test_array WHERE i @> '{}';
SELECT * FROM test_array WHERE i @> '{1}';
SELECT * FROM test_array WHERE i @> '{2}';
SELECT * FROM test_array WHERE i @> '{3}';
SELECT * FROM test_array WHERE i @> '{4}';
SELECT * FROM test_array WHERE i @> '{1,2,4}';
SELECT * FROM test_array WHERE i @> '{1,2,3,4}';
SELECT * FROM test_array WHERE i @> '{4,3,2,1}';
SELECT * FROM test_array WHERE i @> '{0,0}';
SELECT * FROM test_array WHERE i @> '{100}';

EXPLAIN (COSTS OFF) SELECT * FROM test_array WHERE i <@ '{}';
SELECT * FROM test_array WHERE i <@ '{}';
SELECT * FROM test_array WHERE i <@ '{1}';
SELECT * FROM test_array WHERE i <@ '{2}';
SELECT * FROM test_array WHERE i <@ '{1,2,4}';
SELECT * FROM test_array WHERE i <@ '{1,2,3,4}';
SELECT * FROM test_array WHERE i <@ '{4,3,2,1}';
SELECT * FROM test_array WHERE i <@ '{0,0}';
SELECT * FROM test_array WHERE i <@ '{100}';

