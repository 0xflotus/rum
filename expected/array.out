set enable_seqscan=off;
CREATE TABLE test_array (
	i int2[]
);
INSERT INTO test_array VALUES ('{}'), ('{0}'), ('{1,2,3,4}'), ('{1,2,3}'), ('{1,2}'), ('{1}');
CREATE INDEX idx_array ON test_array USING rum (i);
EXPLAIN (COSTS OFF) SELECT * FROM test_array WHERE i = '{}';
                QUERY PLAN                
------------------------------------------
 Index Scan using idx_array on test_array
   Index Cond: (i = '{}'::smallint[])
(2 rows)

SELECT * FROM test_array WHERE i = '{}';
 i  
----
 {}
(1 row)

SELECT * FROM test_array WHERE i = '{0}';
  i  
-----
 {0}
(1 row)

SELECT * FROM test_array WHERE i = '{1}';
  i  
-----
 {1}
(1 row)

SELECT * FROM test_array WHERE i = '{1,2}';
   i   
-------
 {1,2}
(1 row)

SELECT * FROM test_array WHERE i = '{2,1}';
 i 
---
(0 rows)

SELECT * FROM test_array WHERE i = '{1,2,3,3}';
 i 
---
(0 rows)

SELECT * FROM test_array WHERE i = '{0,0}';
 i 
---
(0 rows)

SELECT * FROM test_array WHERE i = '{100}';
 i 
---
(0 rows)

EXPLAIN (COSTS OFF) SELECT * FROM test_array WHERE i && '{}';
                QUERY PLAN                
------------------------------------------
 Index Scan using idx_array on test_array
   Index Cond: (i && '{}'::smallint[])
(2 rows)

SELECT * FROM test_array WHERE i && '{}';
 i 
---
(0 rows)

SELECT * FROM test_array WHERE i && '{1}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
 {1,2}
 {1}
(4 rows)

SELECT * FROM test_array WHERE i && '{2}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
 {1,2}
(3 rows)

SELECT * FROM test_array WHERE i && '{3}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
(2 rows)

SELECT * FROM test_array WHERE i && '{4}';
     i     
-----------
 {1,2,3,4}
(1 row)

SELECT * FROM test_array WHERE i && '{1,2}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
 {1,2}
 {1}
(4 rows)

SELECT * FROM test_array WHERE i && '{1,2,3}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
 {1,2}
 {1}
(4 rows)

SELECT * FROM test_array WHERE i && '{1,2,3,4}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
 {1,2}
 {1}
(4 rows)

SELECT * FROM test_array WHERE i && '{4,3,2,1}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
 {1,2}
 {1}
(4 rows)

SELECT * FROM test_array WHERE i && '{0,0}';
  i  
-----
 {0}
(1 row)

SELECT * FROM test_array WHERE i && '{100}';
 i 
---
(0 rows)

EXPLAIN (COSTS OFF) SELECT * FROM test_array WHERE i @> '{}';
                QUERY PLAN                
------------------------------------------
 Index Scan using idx_array on test_array
   Index Cond: (i @> '{}'::smallint[])
(2 rows)

SELECT * FROM test_array WHERE i @> '{}';
     i     
-----------
 {}
 {0}
 {1,2,3,4}
 {1,2,3}
 {1,2}
 {1}
(6 rows)

SELECT * FROM test_array WHERE i @> '{1}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
 {1,2}
 {1}
(4 rows)

SELECT * FROM test_array WHERE i @> '{2}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
 {1,2}
(3 rows)

SELECT * FROM test_array WHERE i @> '{3}';
     i     
-----------
 {1,2,3,4}
 {1,2,3}
(2 rows)

SELECT * FROM test_array WHERE i @> '{4}';
     i     
-----------
 {1,2,3,4}
(1 row)

SELECT * FROM test_array WHERE i @> '{1,2,4}';
     i     
-----------
 {1,2,3,4}
(1 row)

SELECT * FROM test_array WHERE i @> '{1,2,3,4}';
     i     
-----------
 {1,2,3,4}
(1 row)

SELECT * FROM test_array WHERE i @> '{4,3,2,1}';
     i     
-----------
 {1,2,3,4}
(1 row)

SELECT * FROM test_array WHERE i @> '{0,0}';
  i  
-----
 {0}
(1 row)

SELECT * FROM test_array WHERE i @> '{100}';
 i 
---
(0 rows)

EXPLAIN (COSTS OFF) SELECT * FROM test_array WHERE i <@ '{}';
                QUERY PLAN                
------------------------------------------
 Index Scan using idx_array on test_array
   Index Cond: (i <@ '{}'::smallint[])
(2 rows)

SELECT * FROM test_array WHERE i <@ '{}';
 i  
----
 {}
(1 row)

SELECT * FROM test_array WHERE i <@ '{1}';
  i  
-----
 {}
 {1}
(2 rows)

SELECT * FROM test_array WHERE i <@ '{2}';
 i  
----
 {}
(1 row)

SELECT * FROM test_array WHERE i <@ '{1,2,4}';
   i   
-------
 {}
 {1,2}
 {1}
(3 rows)

SELECT * FROM test_array WHERE i <@ '{1,2,3,4}';
     i     
-----------
 {}
 {1,2,3,4}
 {1,2,3}
 {1,2}
 {1}
(5 rows)

SELECT * FROM test_array WHERE i <@ '{4,3,2,1}';
     i     
-----------
 {}
 {1,2,3,4}
 {1,2,3}
 {1,2}
 {1}
(5 rows)

SELECT * FROM test_array WHERE i <@ '{0,0}';
  i  
-----
 {}
 {0}
(2 rows)

SELECT * FROM test_array WHERE i <@ '{100}';
 i  
----
 {}
(1 row)

